### PR TITLE
Don't export variables from case statements

### PR DESCRIPTION
**Pull request by reiddraper**
I temporarily used the `warn_export_vars` compile option to spot these. I've also spotted a few areas where we repeat ourselves, but I think it'd be best to tackle that in a separate issue. Or a separate commit on this same branch.

### CODE REVIEW

File: src/riak_cs_access_archiver.erl [hunk 1]
```diff
@@ -185,18 +185,8 @@ riak_connection() ->
...
-        undefined ->
-            Host = "127.0.0.1"
-    end,
-    case application:get_env(riak_cs, riak_pb_port) of
-        {ok, Port} ->
-            ok;
-        undefined ->
-            Port = 8087
-    end,
+    Host = application:get_env(riak_cs, riak_ip, "127.0.0.1"),
```
Comment (kellymclaughlin) [inline src/riak_cs_access_archiver.erl:188]: How about making some functions in `riak_cs_config` for these?

Comment (kellymclaughlin) [inline src/riak_cs_access_archiver.erl:188]: Or even a single function that returns `{Host, Port}`

File: src/riak_cs_app.erl [hunk 2]
```diff
@@ -128,16 +128,6 @@ riak_connection() ->
...
-        undefined ->
-            Host = "127.0.0.1"
-    end,
-    case application:get_env(riak_cs, riak_pb_port) of
-        {ok, Port} ->
-            ok;
-        undefined ->
-            Port = 8087
-    end,
+    Host = application:get_env(riak_cs, riak_ip, "127.0.0.1"),
```
Comment (kellymclaughlin) [inline src/riak_cs_app.erl:131]: `application:get_env/3` only appeared in R16 so builds with R15 fail at start up

**Comment by kellymclaughlin** [comment]
- [x] Eunit tests
- [x] EQC tests
- [x] Integration tests
- [x] Client tests
- [ ] Dialyzer output clean
- [ ] xref output
- [x] riak_test run

**Comment by kellymclaughlin** [comment]
Should we use `warn_export_vars` by default?

**Comment by reiddraper** [comment]
> Should we use warn_export_vars by default?

A bunch of our dependencies fail with this on. Maybe there's some way we can turn it on for CS, but not our dependencies?

File: src/riak_cs_app.erl (hunk 2 above)
Comment (reiddraper) [inline src/riak_cs_app.erl:131]: Yeah, I was thinking we were going all-in for R16 for the next version, but maybe instead we should DRY-up all of the times we use this and use the case statement there, so we can be backward compatible.

File: src/riak_cs_access_archiver.erl (hunk 1 above)
Comment (reiddraper) [inline src/riak_cs_access_archiver.erl:188]: +1, and that way we can avoid using get_env/3 in multiple places. Just use get_env/2 once.

**Comment by kuenishi** [comment]
Note: after #848 merged, pushing any more commits without rebasing to develop  or running CI tests will break the plt file in our CI environment. Please rebase to develop, or branch including #848 .

**Comment by kuenishi** [comment]
Moving to 1.5.1 as cleaning up the code is good, while forcing `warn_export_vars` would be another issue (it's bigger fish to fry).

**Comment by reiddraper** [comment]
Doesn't merge cleanly. Closing.

**Pull request closed without merging**